### ISSUE TITLE
Add support for authSource connection string option

### ISSUE THREAD

**Issue by Scorpiion** [mongo-dart/mongo_dart#72]
Connection string options are not supported at the moment, I would really like to have support for the authSource option.

Example: 
```dart
  var mongodbUri = 'mongodb://user:pass@localhost/my_test_db?authSource=admin';
```

The "authAdmin" option makes it easier to centralize user management in a single database instead of creating users for/inside each database (users are created in admin but are given roles on other databases).

To test this, create a new user like this (copy paste when logged in with admin rights to mongodb shell):
```
use admin
db.createUser(
  {
    user: "dbuser",
    pwd: "dart",
    roles: [ { role: "readWrite", db: "my_test_db" } ]
  }
)
```

And after that it should be possible to connect and authenticate with this connection URI: 
```dart
var uri = 'mongodb://dbuser:dart@localhost/my_test_db?authSource=admin';
```

From the command line the connection command looks like this (run "use my_test_db" after connected): 
```dart
mongo --authenticationDatabase admin --username dbuser --password dart localhost
```

I have tried to look at the code how to add this but have not completely figured it out, at first we need to save the options part of the uri on the ServerConfig object. That should be easy inside _parseUri() since we can use 'uri.queryParameters["options"]' to get the options string parsed and ready to go. But after that I'm not sure where it should be passed on to the server or how to do that part... Do you think it would be rather simple to use that string or is it complicated @vadimtsushko?

Read more about connection string options here:
- http://docs.mongodb.org/manual/reference/connection-string/#connections-connection-options

**Comment by vadimtsushko** [comment]
Hi, Robert.
I'll look at this. Can not give a concrete time when though, I'm a little time strapped right now

**Comment by Scorpiion** [comment]
Okay, I understand, let me know if there is something I can do to help!

**Comment by vadimtsushko** [comment]
Hi, Robert. Last commit added authSource connection option to current (dev) version.
Could you check it on your scenarios?

**Comment by Scorpiion** [comment]
Hi Vadim, I have tried this for my scenario and it works good. I don't follow fully on the testing, but maybe you mean because current testing uses mongolab and they simply don't use authSource? For a local mongodb instance that you control it should not be any harder than writing tests involving auth in general I think. 

Either way I think the code needed for this looks pretty straight forward, if normal auth works authSource should work as well, not a huge problem if we don't have an explicit test at this point I think.

**Comment by vadimtsushko** [comment]
it's landed in 0.2.5-beta
